### PR TITLE
Bugfix 1095 use ocvsift

### PR DESCRIPTION
**Pull request by tomas789**
This PR is addressing those issues [Crash: Extracting OpenCV features.](https://github.com/openMVG/openMVG/issues/1097), [Compilation failure: OpenCV SIFT: `Allocate` has incorrect return type.](https://github.com/openMVG/openMVG/issues/1096) and [USE_OCVSIFT is not defined for main_ComputeFeatures_OpenCV](https://github.com/openMVG/openMVG/issues/1095)

### CODE REVIEW

**Comment by tomas789** [comment]
Is there a way how to check what happened with CI?

File: src/software/SfM/main_ComputeFeatures_OpenCV.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 // The <cereal/archives> headers are special and must be included first.
 #include <cereal/archives/json.hpp>
 
+#include "openMVG/features/regions_factory_io.hpp"
 #include "openMVG/image/image_io.hpp"
```
Comment (pmoulon) [inline src/software/SfM/main_ComputeFeatures_OpenCV.cpp]: Please order the include in alphabetical order.

**Comment by pmoulon** [comment]
Travis and Appveyor CI check did not run and I did not know why.
Usually they run on every PR fine.

File: src/software/SfM/main_ComputeFeatures_OpenCV.cpp (hunk 1 above)
Comment (tomas789) [reply src/software/SfM/main_ComputeFeatures_OpenCV.cpp]: @pmoulon Resolved. I did not realise you are sorting them by the name of a file. I'm usually sorting them by whole path.

**Comment by pmoulon** [comment]
I think integration test are not running since there is conflicts so the branch cannot be merged and so Continuous integration test are not run.

File: src/software/SfM/CMakeLists.txt [hunk 2]
```diff
@@ -338,7 +338,9 @@ IF(OpenMVG_USE_OPENCV)
     vlsift
     ${OpenCV_LIBS})
   target_include_directories(openMVG_main_ComputeFeatures_OpenCV PRIVATE ${OpenCV_INCLUDE_DIRS})
-  target_compile_definitions(openMVG_main_ComputeFeatures PRIVATE USE_OCVSIFT)
+  IF(OpenMVG_USE_OCVSIFT)
+    target_compile_definitions(openMVG_main_ComputeFeatures_OpenCV PRIVATE USE_OCVSIFT=1)
```
Comment (pmoulon) [inline src/software/SfM/CMakeLists.txt]: Does the =1 is required?
`target_compile_definitions(openMVG_main_ComputeFeatures_OpenCV PRIVATE USE_OCVSIFT)` should work

Comment (pmoulon) [reply src/software/SfM/CMakeLists.txt]: @tomas789 Can you test with just `target_compile_definitions(openMVG_main_ComputeFeatures_OpenCV PRIVATE USE_OCVSIFT)`

**Comment by tomas789** [comment]
Yes, it works even without it.

**Comment by pmoulon** [comment]
Thanks. Modification have been merged

**Comment by mrliujie** [comment]
I have changed my code modified by "Minor fix following #1098". But when i run throught this programme, it catch an exception in 
 ```archive(cereal::make_nvp("image_describer", image_describer));```
The output is shown as followed 
```
Trying to load an unregistered polymorphic type (SIFT_Image_describer).
Make sure your type is registered with CEREAL_REGISTER_TYPE and that the archive you are using was included (and registered with CEREAL_REGISTER_ARCHIVE) prior to calling CEREAL_REGISTER_TYPE.
If your type is already registered and you still see this error, you may need to use CEREAL_REGISTER_DYNAMIC_INIT.
```
I suppose it may be something wrong in 
```
CEREAL_REGISTER_TYPE_WITH_NAME(SIFT_OPENCV_Image_describer, "SIFT_OPENCV_Image_describer");
CEREAL_REGISTER_POLYMORPHIC_RELATION(openMVG::features::Image_describer, SIFT_OPENCV_Image_describer)
```
,  but i can't fix it.
Can you help me to figure it out?

Cheers,
Jack

**Comment by pmoulon** [comment]
This issue should be fixed in the develop branch. Can you try to use the develop branch?

**Pull request closed without merging**